tests/devlxd-vm

```diff
@@ -90,14 +95,47 @@ fi
 # Wait for boot to be completed to interact with snap/snapd
 waitInstanceBooted v1
 
+# Below tests for image export rely on an event being emitted on export. This will not be backported to 4.0 or 5.0,
+# so exit the test here.
+if [ "${LXD_SNAP_CHANNEL}" = "4.0/edge" ] || [ "${LXD_SNAP_CHANNEL}" = "5.0/edge" ]; then
+  echo "Skipping devlxd image export tests, not supported"
+  FAIL=0
+  exit
+fi
+
 # Test nested VM functionality.
 lxc exec v1 -- snap remove --purge lxd || true
 lxc exec v1 -- snap install lxd --channel="${LXD_SNAP_CHANNEL}"
 lxc exec v1 -- /snap/bin/lxd init --auto
+if [ -n "${LXD_SIDELOAD_PATH:-}" ]; then
+    lxc file push "${LXD_SIDELOAD_PATH}" v1/var/snap/lxd/common/lxd.debug
+    lxc exec v1 -- systemctl restart snap.lxd.daemon
+fi
+if [ -n "${LXD_AGENT_SIDELOAD_PATH:-}" ]; then
+  lxc file push "${LXD_AGENT_SIDELOAD_PATH}" "v1/root/$(basename "${LXD_AGENT_SIDELOAD_PATH}")"
+  lxc exec v1 -- mount --bind "$(basename "${LXD_AGENT_SIDELOAD_PATH}")" /snap/lxd/current/bin/lxd-agent
+  lxc exec v1 -- systemctl restart snap.lxd.daemon
+fi
+
+monitorPID=""
+if hasNeededAPIExtension devlxd_images && hasNeededAPIExtension event_lifecycle; then
```

Comment:
LXD 4.0 supports both `devlxd_images` and `event_lifecycle` so maybe we could drop the conditional altogether?